Added a way to disable the clr object

C0bra5 opened the pull request:
Summary of changes:
For security reasons, I need a way to disable the CLR functionality. To achieve this goal, I added a new parameter to the `LuaCompileOptions` class called `ClrEnabled`. To preserve existing functionality, `ClrEnabled` is set to true by default. I've also added ways to test this new functionality through the unit tests and via the `NeoCmd` utility: I've also added a  new section at the bottom of the CLR documentation file to explain how disabling CLR works.

List of changes:
* Added a new parameter called `ClrEnabled` to the `LuaCompileOptions` class (set to true by default).
* Added a check for the `ClrEnabled` option in the `ParsePrefix` function of the `Parser` class.
* Added `TestClrEnabled` and `TestClrDisabled` in the `Parser.cs` unit test file.
* Added the `:clron` and `clroff` to the `NeoCmd` utility.
* Added documentation for the new `NeoCmd` commands.
* Added documentation for the new `ClrEnabled` parameter in the `CLR` documentation file.

C0bra5 committed:
Added a way to dissable the clr object

Summary of changes:
For security reasons, I need a way to disable the
CLR functionality. To achieve this goal, I added a new parameter to the
`LuaCompileOptions` class called `ClrEnabled`. To preserve existing
functionality, `ClrEnabled` is set to true by default. I've also added
ways to test this new functionality through the unit tests and via the
`NeoCmd` utility: I've also added a  new parameter at the bottom of the
CLR documentation file to explain how disabling CLR works.

List of changes:
* Added a new parameter called `ClrEnabled` to the `LuaCompileOptions`
class (set to true by default).
* Added a check for the `ClrEnabled` option in the `ParsePrefix`
function of the `Parser` class.
* Added `TestClrEnabled` and
`TestClrDisabled` in the `Parser.cs` unit test file.
* Added the
`:clron` and `clroff` to the `NeoCmd` utility.
* Added documentation for the new `NeoCmd` commands;
* Added documentation for the new `ClrEnabled` parameter in the `CLR`
documentation file.

C0bra5 committed:
why did this not get checked in...

Somehow this change wasn't check in

neolithos commented on NeoCmd/Program.cs:
```diff
@@ -196,6 +198,7 @@ protected override void OnFrameExit()
 		private static Lua lua = new Lua(); // create lua script compiler
 		private static LuaGlobal global;
 		private static ILuaDebug debugEngine = LuaStackTraceDebugger.Default;
+		private static bool ClrEnabled = true;
```
Please, can refactor to `clrEnabled`

C0bra5 replied:
class members starting with the lower case...
a) Why?
b) Fixed as requested in 08a5ff2cf6998cbd2131c649f34552d456c7ad69

neolithos commented:
Overall good job 👍

C0bra5 committed:
Fixes naming convention in NeoCmd

Refactores the name of the `ClrEnabled` static variable in the
program.cs of the NeoCmd `clrEnabled` as requested in
https://github.com/neolithos/neolua/pull/116#discussion_r453060943

C0bra5 committed:
removed broken from documentation

Removed a line out of the `Disabling clr access` documentation section.
This permits the code to be dropped in for testing purposes.

neolithos commented:
Do you need a release to nuget?

C0bra5 commented:
My stuff is nowhere near release so feel free to do it whenever you want.

codecat commented:
Unfortunately this does not provide a lot of security, because calling into CLR is still possible:
```lua
'':GetType().Assembly:GetType('System.Something.Something'):GetMethod('Something'):Invoke(...)
```

neolithos commented:
Good hint. But it should filterable via `SandBox`?

codecat commented:
True! Something like this seems to suffice (requires tweaking depending on needs):
```cs
DynamicSandbox = a => {
	switch (a) {
		case LuaMethod:
		case string:
			return a;
	}
	return null;
},
```

neolithos commented:
Something like that, but I had never to thing about these kind of problems.
